### DISCUSSION TITLE
new effects, discuss and vote?

### DISCUSSION THREAD

**Discussion by jgarza9788** [Schneegans/Burn-My-Windows#457] [Ideas]
this is just me messing around in unity's shader graph to experiment with and see what looks cool.

https://github.com/user-attachments/assets/34805fc0-68b7-45f0-958b-383b31e2fd0d

1. Apple Siri New Effect Thingie
try to make it look like the new Apple siri feature, that color changes at the edge of the window.

2. liquid fill in
using some noise to make it look like liquid is filling the window

3. RGB glitch
the TV glitch is nice, but we can rip the RGB channels apart and spread them out with some noise to make a slightly different glitch

4. mushroom
this one would only make sense to the super mario lovers.
but basically growing and shrinking the window as if it got a mushroom

5. Color SVGish Fire smoke thingie
i have no idea what i was trying to go for here, but it looks cool.

:tada: :tada: :tada: :tada: :tada:

**Comment by Schneegans** [comment]
I would say: go for it! The existing glitch already distorts RGB differently, but we can also create more glitch variants, I suppose :smile: 

I really like the mushroom :mushroom: 

Btw, as long as an effect does not require an additional input texture, we can also port it to KDE.

**Comment by jgarza9788** [comment]
Mushroom coming up ...
and some people say that i'm not a **fungi **

https://github.com/user-attachments/assets/2b26d96c-4874-4e08-a093-f14e85a6209e

**Reply by jgarza9788** [reply]
and it would go great with this window style

![super_mario_theme_by_necro949445_d1t7unt-fullview](https://github.com/user-attachments/assets/0fd6422e-9e09-4e19-b564-f5474fca8f19)
https://www.deviantart.com/necro949445/art/super-mario-theme-109541369

**Comment by jgarza9788** [comment]
WIP... but i hope you like this update
i'll have a switch for 8bit and ...idk "modern"?

https://github.com/user-attachments/assets/d63a06c5-03cf-46e9-986c-4d297620a642

**Reply by Schneegans** [reply]
That's plain awesome! Are the stars procedurally generated or based on a texture?

**Reply by jgarza9788** [reply]
procedurally generated

**Reply by Schneegans** [reply]
Do you have an ETA when this will be ready? I soon would like to release a new version of the extension. Should I wait for this or do we want to include this in a later release?

**Reply by jgarza9788** [reply]
No, I don't have an eta...
If we rush it, it might look bad, but it's just an effect.

**Reply by Schneegans** [reply]
Yeah, don't rush it. Then I'll release a new version rather soonishly and we'll add any further effects later on.

**Reply by jgarza9788** [reply]
not ready yet... but getting there 
oh and i should clean the code a bit before i submit
...but check it out

https://github.com/user-attachments/assets/17b1cd74-a17f-40e6-9b51-7e4cb955c673

**Comment by jgarza9788** [comment]
it's been a while since my last update

![image](https://github.com/user-attachments/assets/a3b5739d-0dbf-46f3-9594-d29882126250)
[x] small 4point stars (sparks???)
[x] those strange vertical lines (calling them rays)
[x] rings of stars ... that change gradient over time

![image](https://github.com/user-attachments/assets/c3731280-53b4-4e0a-a040-c72ccb27a1b1)
[x] everything disables when using the 8bit style mode...

i want to clean up the GLSL a bit...
but it's coming along 

suggestions?

**Reply by Schneegans** [reply]
🤯

**Comment by jgarza9788** [comment]
https://youtu.be/T_avmeAQA4k

**Reply by Schneegans** [reply]
That's totally awesome. I love the insane overkill in configuration options for an effect most people will never consider using as it's so annoying. But it just shows off how fun it is :smile:

**Reply by jgarza9788** [reply]
i like the the look of ring of 5p stars only (no sparks, no rays)
i think if people don't do the 8bit... they'd do that.

and this siri-ish one can be next!
https://www.shadertoy.com/view/4ftBWH

**Reply by jgarza9788** [reply]
ok, i think i need your help...
I'm trying to do a clean branch for you to merge.
and when doing a test compile ... it's not moving the Mushroom.js to the .local/.../src/effects folder.

![image](https://github.com/user-attachments/assets/7f496166-ff5e-4e51-9938-bd12f5f6b708)

**Reply by jgarza9788** [reply]
nevermind... it's working now.
continuing testing 😅

**Comment by jgarza9788** [comment]
AuraGlow (Coming Soon)
https://github.com/user-attachments/assets/28cb6611-934b-41f1-8961-60b3a3962bd3

**Reply by jgarza9788** [reply]
https://youtu.be/W5kbCBasBx8

**Reply by Schneegans** [reply]
Very cool!

I think the only improvement I could potentially see is that the shrinking cut-off border looks very sharp. I don't know what would be the best way to improve this. Maybe simply not a 100% sharp cut off? More like a few pixel wide gradient? Or maybe it would already look a bit better if the window would fade to transparency?

Also, these color effects would make for a cool addition to the "Glide" effect. Like a new "Aurora Glide" effect!

**Reply by jgarza9788** [reply]
## i have blur bleed 
... if true the blur will not (for the lack of a better term) respect the edge of the window.
... if false the edge of the window will be respected.

## fade with transparency
https://github.com/jgarza9788/Burn-My-Windows/blob/AuraGlow_WIP/resources/shaders/aura-glow.frag
```glsl
    //i want to fade out the last ~10% of the animation
    float lastfade = remap(progress,0.0,0.1,0.0,1.0);
    lastfade = clamp(lastfade,0.0,1.0);
    lastfade = easeInSine(lastfade);
    
    //apply the lastfade
    oColor.a *= lastfade;
````    
 ^i can turn this so people can choose the last X should be a transparent fade
 
## Aurora Glide
should it be added to this, or a new effect ?
- but i agree it could look very cool

## one more thing
this is a small thing, but i think i need to adjust so that the light and dark modes both look good.
the light mode doesn't look like the color that is selected in the menu (it wraps around because the white is already too bright)
... if i can't think of a fix it's not the end of the world, but it would be nice.

---

thanks for the suggestions
this is refreshing to work on, my work is boring programming and this has got me excited.
also you are a pleasure to work with.

**Reply by jgarza9788** [reply]
```glsl
    //i want to fade out the last ~10% of the animation
    float lastfade = remap(progress,0.0,uFadeOut,0.0,1.0);
    lastfade = clamp(lastfade,0.0,1.0);
    lastfade = easeInSine(lastfade);
    
    //apply the lastfade
    oColor.a *= lastfade;
```

now the fade-out is adjustable by the user

---

i tried the to adjust with the light/dark mode... it didn't really work out the way i planned 
...maybe i'll try again.

**Reply by jgarza9788** [reply]
getting a second opinion
https://www.reddit.com/r/gnome/comments/1h6e8zh/working_on_a_burnmywindows_effect/

**Comment by jgarza9788** [comment]
AuraGlow_Clean branch
https://github.com/jgarza9788/Burn-My-Windows/tree/AuraGlow_clean

**Comment by jgarza9788** [comment]
~4 more maybe...
1. Warp ? ...  but maybe it should be from the edge to the center ?
https://www.shadertoy.com/view/4cdfWl

2. Star ... basically just use a star as a mask for the window
https://www.shadertoy.com/view/MctfWS

3. Freeze 
https://www.shadertoy.com/view/4fVBzz

4. Unicorn Fart (mostly for the shitz and giggles)
... no link

**Reply by Schneegans** [reply]
I think they all are very promising. Warp looks cool, Star has some Christmas vibes to it. Freeze looks cool as well, however I expected something more _shock-frosting like_. Maybe like the window turning into ice starting from the mouse pointer. With cracks and crystals forming in a fractal manner and then falling into pieces :smile: 

Unicorn fart could be a mixture between Focus and Aurora, I guess!
